c2rust-transpile/src/translator/mod.rs

```diff
@@ -376,17 +376,13 @@ fn vec_expr(val: Box<Expr>, count: Box<Expr>) -> Box<Expr> {
 }
 
 pub fn stmts_block(mut stmts: Vec<Stmt>) -> Box<Block> {
-    if stmts.len() == 1 {
-        if let Stmt::Expr(ref e) = stmts[0] {
-            if let Expr::Block(ExprBlock {
-                block: ref b,
-                label: None,
-                ..
-            }) = e
-            {
-                return Box::new(b.clone());
-            }
-        }
+    match stmts.as_slice() {
+        [Stmt::Expr(Expr::Block(ExprBlock {
+            block: b,
+            label: None,
+            ..
+        }))] => return Box::new(b.clone()),
+        _ => {}
     }
```

Comment:
This seems better written as an `if let` with a slice pattern, no? I would expect the `single_match` lint to complain about this.

Reply:
It does, that's why it's silenced 😕 But imho using if-let with huge patterns is very clunky. It also somewhat obscures the scrutinee, and causes an inversion of flow (pattern before the match value) which I find confusing for big patterns.

However, I agree that it's more of a my personal preference in this specific case than some overarching policy. I have changed some matches into if-lets with big patterns in this PR. I think this specific form looks better, but if you would insist on consistency I can change it into an if-let.

I'm not a fan of huge patterns, but I don't have a better solution.

Reply:
Wait, that's not equivalent.  I'll figure it out.

Reply:
I don't think such significant restructuring of a function fits in a PR with mostly automatically applicable changes. If you want to change it, it would be better to make it a separate PR.

There are, in fact, a lot of such functions which could benefit from some cleanup.

Reply:
Hmm, I did clean up a few other functions already.  You're probably right that they should go in separate PRs, but that also makes it more work and takes longer to separate everything out.

For this one and any similar ones, how about we revert them to whatever is fully `clippy`-approved and then we can refactor them later in a separate PR to make them cleaner.